R/zchunk_LA154.Transport_china.R

```diff
@@ -75,38 +79,28 @@ module_gcam.china_LA154.Transport <- function(command, ...) {
         ungroup() ->
         L154.NBS_trn_Mtce_province
 
-      # To calculate the province share, we need to calculate the national amount
-      L154.NBS_trn_Mtce_province %>%
-        group_by(EBProcess, EBMaterial, year) %>% # Dropping province
-        summarise(value_national = sum(value, na.rm = T)) %>%
-        ungroup() ->
-        NBS_transportation_national
-
-      # Now the province shares can be calculated by dividing the province data by the national
+      # Now the province shares can be calculated
       L154.NBS_trn_Mtce_province %>%
-        left_join_error_no_match(NBS_transportation_national, by = c("EBProcess", "EBMaterial", "year")) %>%
-        mutate(value_share = value / value_national) %>% # Calculating province's share
+        group_by(EBProcess, EBMaterial, year) %>%
+        mutate(value_share = value / sum(value, na.rm = T)) %>%
+        complete(nesting(year, sector, fuel, EBProcess, EBMaterial), province = gcamchina.PROVINCES, fill = list(value_share = 0)) %>%
         # NAs were introduced where national values were 0. Replace NAs with zeros.
         replace_na(list(value_share = 0)) %>%
         select(province, EBProcess, EBMaterial, year, value_share) ->
         L154.NBS_trn_share_province
 
-      # Match these percentages into a table of all transportation technologies that will be written out
-      L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>%
-        repeat_add_columns(tibble::tibble(province = gcamchina.PROVINCES)) ->
-      L154.pct_province_trn_m_sz_tech_F_Yh
 
-      # TODO: Tibet
-      #L154.pct_province_trn_m_sz_tech_F_Yh <- subset( L154.pct_province_trn_m_sz_tech_F_Yh, province != "XZ" )
-
-      L154.pct_province_trn_m_sz_tech_F_Yh %>%
+      L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>%
         filter(year %in% HISTORICAL_YEARS) %>%
         mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>%
-        left_join(L154.NBS_trn_share_province, by = c("province", "EBProcess", "EBMaterial", "year")) %>%
+        full_join(L154.NBS_trn_share_province, by = c("EBProcess", "EBMaterial", "year")) %>%
         replace_na(list(value_share = 0)) %>%
         mutate(value = NULL) ->
```

Comment:
I think this is the national value, why drop and join the data back later? Why not multiply value and share directly?